### PR TITLE
bsunanda:Phase2-hgx57 Bug fixing of the geometry loader and validation code

### PR DESCRIPTION
**Pull request by bsunanda**
Resolved the issue of z position of the BH in Reco geometry
Resolved the issue of matching SIM/REC level position of hits

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @dmitrijus @davidlange6 Could you kindly approve the PR?

**Comment by dmitrijus** [comment]
+1

**Comment by ianna** [comment]
+1

File: Validation/HGCalValidation/test/HGCHitValidation.cc [hunk 1]
```diff
@@ -74,46 +74,53 @@
 #include <TH1.h>
 #include <TH2.h>
 #include <TH3.h>
+#include <TVector3.h>
```
Comment (davidlange6) [inline Validation/HGCalValidation/test/HGCHitValidation.cc]: @bsunanda - why is this needed?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@dmitrijus @davidlange6 I resubmitted the PR after discarding a few unused includes and changing a few histogram ranges

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@dmitrijus @davidlange6 Could you approve and get it integrated

**Comment by bsunanda** [comment]
@ianna Could you approve this yet again. Did some clean up as suggested by David

**Comment by dmitrijus** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by davidlange6** [comment]
+1
